pkg/reconciler/pipeline/dag/dag.go

```diff
@@ -163,9 +200,7 @@ func addLink(pt string, previousTask string, nodes map[string]*Node) error {
 		return fmt.Errorf("task %s depends on %s but %s wasn't present in Pipeline", pt, previousTask, previousTask)
 	}
 	next := nodes[pt]
```

Comment:
This is holding a huge struct (object) in real world applications. It is set to the entire `pipelineTask` specification along with a list of `params`, `when` expressions, the entire specification when `taskSpec` is specified, etc. The `pipelineTask` specification is not required at this point since all the dependencies (`runAfter`, task results in params and when expressions, and from) are calculated before calling `dag.Build`. All we need is just the `pipelineTask.Name` i.e. `HashKey()`.  We have a room of improvement in future to avoid passing the entire blob around.